go/api/commands.go

```diff
@@ -155,3 +155,230 @@ type StringCommands interface {
+	//  false if field already exists in the hash and no operation was performed.
+	//
+	// For example:
+	//  payload1, err := client.HSetNX("myHash", "field", "value")
+	//  // payload1 equals true
+	//
+	//  payload2, err := client.HSetNX("myHash", "field", "newValue")
+	//  // payload2 equals false
+	//
+	// [valkey.io]: https://valkey.io/commands/hsetnx/
+	HSetNX(key, field, value string) (bool, error)
+
+	// HDel removes the specified fields from the hash stored at key.
+	// Specified fields that do not exist within this hash are ignored.
+	// If key does not exist, it is treated as an empty hash and this command returns 0.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key    - The key of the hash.
+	//  fields - The fields to remove from the hash stored at key.
+	//
+	// Return value:
+	//  The number of fields that were removed from the hash, not including specified but non-existing fields.
+	//
+	// For example:
+	//  num, err := client.HDel("my_hash", "field1", "field2")
+	//  // num equals 2
+	//
+	// [valkey.io]: https://valkey.io/commands/hdel/
+	HDel(key string, fields ...string) (int64, error)
+
+	// HLen returns the number of fields contained in the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key - The key of the hash.
+	//
+	// Return value:
+	//  The number of fields in the hash, or 0 when key does not exist.
+	//  If key holds a value that is not a hash, an error is returned.
+	//
+	// For example:
+	//  num1, err := client.HLen("myHash")
+	//  // num1 equals 3
+	//
+	//  num2, err := client.HLen("nonExistingKey")
+	//  // num2 equals 0
+	//
+	// [valkey.io]: https://valkey.io/commands/hlen/
+	HLen(key string) (int64, error)
+
+	// HVals returns all values in the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key - The key of the hash.
+	//
+	// Return value:
+	//  A slice of strings containing all the values in the hash, or an empty slice when key does not exist.
+	//
+	// For example:
+	//  values, err := client.HVals("myHash")
+	//  // values equals []string{"value1", "value2", "value3"}
+	//
+	// [valkey.io]: https://valkey.io/commands/hvals/
+	HVals(key string) ([]string, error)
+
+	// HExists returns if field is an existing field in the hash stored at key.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  key   - The key of the hash.
+	//  field - The field to check in the hash stored at key.
+	//
+	// Return value:
+	//  true if the hash contains the specified field.
+	//  false if the hash does not contain the field, or if the key does not exist.
+	//
+	// For example:
+	//  exists, err := client.HExists("my_hash", "field1")
+	//  // exists equals true
+	//
+	//  exists, err = client.HExists("my_hash", "non_existent_field")
+	//  // exists equals false
+	//
+	// [valkey.io]: https://valkey.io/commands/hexists/
+	HExists(key, field string) (bool, error)
```

Comment:
You omit key type somewhere, but I'd not. At least align it for all commands.

Reply:
@Yury-Fridlyand It's done 🚀 .